src/app/ReadClient.h

```diff
@@ -110,11 +110,15 @@ class ReadClient : public Messaging::ExchangeDelegate
          * receives an OnDone call to destroy the object.
          *
          * @param[in] aPath        The attribute path field in report response.
+         * @param[in] aVersion     The data version for cluster in report response. The version could be omitted when
+         *                         EnableTagCompression in the Path field is true.
```

Comment:
Tag compression is an over the wire detail. By the time it gets up here to this call-site, we should _**always**_ have a data version on hand. Consequently, this should not be an `Optional`.

Reply:
Done, thanks